### PR TITLE
docs: upgrading to v3.3

### LINKED ISSUE #4017: docs: upgrade guide from v3.2 to v3.3
Steps:
1- Cookiecut an instance. 
2- Edit the Pipfile to use the Invenio v3.2.2 module. Install from git or local checkout of https://github.com/inveniosoftware/invenio/pull/4015
3- Follow the quickstart (Okay you already cookiecutted, but you have to bootstrap, setup and server)
4- Create some records and so on. Follow: https://invenio.readthedocs.io/en/latest/quickstart/crud-operations.html
5- Try to upgrade the instance to v3.3. Basically you should change the invenio version in the Pipfile. In this case use: https://github.com/inveniosoftware/invenio/pull/4013. You might have to do extra steps (e.g. remove lxml restriction, etc.)

The task consists on documenting every singly change that needs to be made to update from v3.2.2 to v.3.3

Similar to https://github.com/inveniosoftware/invenio/blob/edb863d2f5228fb158c090a69c2db7c3385b6ba3/docs/upgrading.rst

### PR DESCRIPTION
**Pull request by diegodelemos**
* Closes #4017.

### CODE REVIEW

**Comment by lnielsen** [comment]
Text is good, I'd however like to keep the existing upgrade docs from 3.1 to 3.2 so that if you're on v3.1 you have both upgrade guides available. The table of contents I would like to change like this:

- Upgrading
  - Upgrading from v3.2 to v3.3
  - Upgrading from v3.1 to v3.2
- Releases
  - Maintenance policy
  - ...

Screenshot 2020-05-15 at 08 29 07

File: docs/upgrading.rst [hunk 1]
```diff
@@ -5,8 +5,63 @@
...
+
+2. Make sure that your database and Elasticsearch version matches your
+   installation. In above example the database is ``postgresql`` and the
+   Elasticsearch version is ``elasticsearch7``.
+
+3. Remove the following line from Pipfile:
+
+.. code:: diff
+
+    - lxml = ">=3.5.0,<4.2.6"
```
Comment (ppanero) [inline docs/upgrading.rst:42]: **moderate**: Remove also `SQLAlchemy-Utils`.

**Comment by zzacharo** [comment]
@ppanero nope I don't think so :)

**Comment by ppanero** [comment]
Closed in favour of #4013

**Pull request closed without merging**